### ISSUE TITLE
TGraphAsymmErrors unusable.

### ISSUE THREAD

**Issue by marinang** [scikit-hep/uproot-methods#6]
Hi,

currently with uproot I cannot do anything with with TGraphAsymmErrors object. Probably TGraph are also not handled by uproot yet (?). Would it be possible the add the treatment of this class ?

Matt

**Comment by jpivarski** [comment]
For _writing_ `TGraph*` objects to ROOT files, that's currently at the bottom of our to-do list. We should handle all the histogram and profile types before graphs because histogram writing methods are generalizable and it will put us into a good pattern for expansion later.

But if you mean _reading_, you should be able to read any `TGraph*` now, though not make use of it (as you said). uproot should give you a Python class object containing a bunch of member data that starts with `_f*`. uproot-methods doesn't yet do any interpretation. I'd need input about what you'd _want_ it to do, though.

[THnSparse](https://github.com/scikit-hep/uproot-methods/blob/master/uproot_methods/classes/THnSparse.py) is an example of a stub for filling out methods of a class, and [TH1](https://github.com/scikit-hep/uproot-methods/blob/master/uproot_methods/classes/TH1.py) is an example of a filled one. Once you've decided how you want a `TGraph*` to appear in Python— what methods you want it to have, how to put `_f*` members to use in Python— it's pretty easy to fill in the class definition.

uproot will automatically pick up the new methods if the module name is equal to the C++ class name and the class is named `Methods` (just like THnSparse). This applies to subclasses as well— notice that we only defined behaviors for `TH1` and all of its subclasses automatically get the same methods. You could put general methods in a `TGraph` module and specialized methods in a `TGraphAsymmErrors` module.

Are you interested in defining this class's behavior?

**Comment by marinang** [comment]
Sorry I didn't precise, but I meant _reading_.

Indeed I found the starting `_f*` and I got the data I wanted thanks. 

Yeah I am interested in defining this class's behavior. It doesn't seem difficult, however for the particular case of `TGraph`,  and then `TGraphErrors`/`TGraphAsymmErrors` the general methods would be in `TGraph.py` and the specialized in `TGraphErrors.py`/`TGraphAsymmErrors.py` like `TH1` and `TH2` right ? 
However I don't see how it is done, i.e, when I read a `TH2` with uproot I do see methods from `TH1` but I cannot where in the repo you say in that `TH2` is a subclass of `TH1`. But maybe you don't say it and this information is given by the streams in the ROOT File and I wouldn't have to worry defining such inheritance relation with `TGraph` and `TGraphErrors`/`TGraphAsymmErrors.
